modules/social_features/social_post/social_post.services.yml

```diff
@@ -11,3 +11,12 @@ services:
   social_post.helper:
     class: Drupal\social_post\Service\SocialPostHelper
     arguments: ['@entity_type.manager', '@current_user']
+
+  #hooks replacement
+  social_post.form.hooks:
+    class: Drupal\social_post\Hooks\SocialPostFormHooks
+    arguments:
+    - '@social_post.helper'
+    - '@current_user'
```

Comment:
See my previous comment about Coding guideline [3.6 Use service autowiring and alias interfaces to concrete ID'd service definitions](https://open-social.slite.com/app/docs/dm052z21PPpHST#2e76f59c) :)

Reply:
My understanding of AutoWiring was erroneous. 
Fixed